wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/Views/DeviceInfoViewController.swift

```diff
@@ -0,0 +1,97 @@
+
+    private var cancellables = Set<AnyCancellable>()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupNavigationItemTitleObservation()
+        setupDebugMenuButtonIfNeeded()
+    }
+
+    private func setupNavigationItemTitleObservation() {
+
+        let certificatePublisher = rootView.viewModel.$e2eIdentityCertificate
+        let isProtuesVerifiedPublisher = rootView.viewModel.$isProteusVerificationEnabled
+        certificatePublisher.combineLatest(isProtuesVerifiedPublisher)
+            .sink { [weak self] certificate, isProteusVerified in
+                self?.updateNavigationItemTitle(certificate, isProteusVerified)
+            }
+            .store(in: &cancellables)
+    }
+
+    private func updateNavigationItemTitle(
+        _ certificate: E2eIdentityCertificate?,
+        _ isProteusVerified: Bool
+    ) {
+
+        let deviceName = NSMutableAttributedString(string: rootView.viewModel.title)
+        if
+            rootView.viewModel.isE2eIdentityEnabled,
+            let certificate,
+            let imageForStatus = certificate.status.uiImage {
+            let attachment = NSTextAttachment(image: imageForStatus)
+            attachment.bounds = .init(origin: .init(x: 0, y: -1.5), size: imageForStatus.size)
+            deviceName.append(.init(string: " "))
+            deviceName.append(.init(attachment: attachment))
+        }
+        if isProteusVerified {
+            let verifiedShield = UIImage(resource: .verifiedShield)
+            let attachment = NSTextAttachment(image: verifiedShield)
+            attachment.bounds = .init(origin: .init(x: 0, y: -1.5), size: verifiedShield.size)
+            deviceName.append(.init(string: " "))
+            deviceName.append(.init(attachment: attachment))
+        }
+
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.attributedText = deviceName
+        label.font = FontSpec(.header, .semibold).font
+        navigationItem.titleView = label
+    }
+
+    private func setupDebugMenuButtonIfNeeded() {
+
+        if rootView.viewModel.isDebugMenuAvailable {
+            let toggleDebugMenu = UIAction(title: "Debug") { [weak self] _ in
+                self?.rootView.viewModel.isDebugMenuPresented.toggle()
+            }
+            let button = UIButton(primaryAction: toggleDebugMenu)
+            button.titleLabel?.font = FontSpec(.normal, .regular).font
+            navigationItem.rightBarButtonItem = .init(customView: button)
+        }
+    }
+}
+
+// `DeviceDetailsView` and `ProfileDeviceDetailsView` are similar and even use the same view model type.
+// The `DeviceInfoView` protocol allows for using the same custom hosting controller for both.
+protocol DeviceInfoView: View {
+    var viewModel: DeviceInfoViewModel { get }
+    init(viewModel: DeviceInfoViewModel)
+}
+extension DeviceDetailsView: DeviceInfoView {}
+extension ProfileDeviceDetailsView: DeviceInfoView {}
```

Comment:
Can the confirmance be moved to the definition of the views instead of extensions?